freetextresponse/public/view.js

```diff
@@ -98,6 +93,16 @@ function FreeTextResponseView(runtime, element) {
         return html;
     }
 
+    function display_responses_if_answered(response) {
```

Comment:
Should I remove this helper function?

Reply:
I wouldn't.

Honestly, I tend to find that anything more than 5 (ish) lines can generally be pulled out into a dedicated function.

Here, this is a great level of abstraction; it's a helper function that makes it abundantly clear what's going on.
Otherwise, you end up needing to overly comment each stanza, something like:

Note: lines with `###` are placeholders for code, while single `#` are actually comments that would be added to the code

```python
def main():
    # initialize a bunch of stuff
    pass  ### ... lots of lines
    
    # fetch data
    pass  ### make ajax calls and whatever

    # parse data
    pass  ### several lines looping over data and processing it

    # display data
    pass  ### all the stuff in `display_responses_if_answered`
```

Now compare that to this version w/ helpers:

```python
def main():
    self.initialize_state()
    data = fetch_data()
    data = parse_data(data)
    display_responses_if_answered(data)
```

Well-named helper functions give us automatic documentation :)

Reply:
Also, see note about JS using camelCase instead of snake_case.

Reply:
Resolved